config_verifier/src/index.ts

```diff
@@ -0,0 +1,389 @@
+            }
+
+            // Chain specific code here for finding the peer that we need
+            var targetPeerAddress;
+
+            if(target['type'] == 'evm'){
+                targetPeerAddress = BigNumber.from(await target['transceivers'][0]['transceiver'].getWormholePeer(src['chainId']));
+            }
+            else if(target['type'] == 'solana'){
+                try{
+                    targetPeerAddress = await target['manager'].program.account.transceiverPeer.fetch(target['manager'].transceiverPeerAccountAddress(src['chainId']))
+                    targetPeerAddress = BigNumber.from("0x" + bufferToHex( targetPeerAddress['address']));
+                }
+                catch(err){
+                    console.log("Error in checkWormholeTransceiver Solana:", err);
+                    targetPeerAddress = BigNumber.from(0x000000000000000000000000000000000000000000000); // Can't find the addresss, since it doesn't exist
+                }
+            }else{
+                console.log("Invaild chain type in checkTransceivers");
+            }
+
+            if(!targetPeerAddress.eq(src['transceivers'][0]['address'])){
+                console.log(`ERROR: Peers don't match chain id for sender chain ${target_id} and receiving chain ${src_id}`);
+            }
+        }
+    }
+    console.log("Finishing transceiver configurations....");
+
+}
+
+function bufferToHex (buffer) {
+    return [...new Uint8Array (buffer)]
+        .map (b => b.toString (16).padStart (2, "0"))
+        .join ("");
+}
+
+/*
+Check the various chain setups corresponding to each other. 
+What is currently checked:
+- Manager peer registration
+- Decimals 
+- Burn/lock bridge configuration
+- Threshold is maxed out
+- Inbound rate limit
+- Outbound rate limit
+*/
+async function checkManagers(chainData){
+
+    console.log("Checking manager configurations....");
+
+    var burn = 0;
+    var lock = 0;
+    for(var target_id in chainData){
+        for (var src_id in chainData){
+
+            var target = chainData[target_id];
+            var src = chainData[src_id];
+
+            if(target['chainId'] == src['chainId']){
+
+                // Mode check
+                if(target['mode'] == 1){
+                    burn += 1;
+                }
+                else if(target['mode'] == 0){
+                    lock += 1;
+                }
+                else{
+                    console.log(`ERROR: Invalid mode for chain ${target_id}`)
+                }
+
+                // Threshold check to ensure its the maximum that it can be. 
+                if(target['threshold'] < target['count']){
+                    console.log(`WARNING: NTT Manager has lower threshold than registered transceivers. ChainID - ${target_id}`); 
+                }
+
+                // Check if the rate limit is turned off or maxed out
+                if(target['outboundRateLimit'].eq(0) || target['outboundRateLimit'].eq(BigNumber.from(2).pow(256).sub(1))){
```

Comment:
I think you can bring the RateLimit checks down by quite a lot:

1. Solana uses u64 to store this value so this check will never trigger even if capacity is at its maximum value 
https://github.com/wormhole-foundation/example-native-token-transfers/blob/0d764bf7ffbbf81b4d51d1bfd5e51929e7fb0460/solana/programs/example-native-token-transfers/src/queue/rate_limit.rs#L8

2. I think it's safe to say that both of these cases would count as "disabled". If the rate limit is set to be the maximum possible integer value, there is no rate limit

3. I wonder if we would add another check that's in the range of 2^32 or even smaller? To me that would still be very high for e.g. a 24h period

Reply:
Yeah! Good ideas. I'm unsure if the `rateLimit` has values that are trimmed or not. I'll confirm this. 

The 'delay' check is good! I'll add that in.